### PR TITLE
[Next release 3.0] Rebrand from AsyncDisplayKit to Texture

### PR DESCRIPTION
**Pull request by Kaspik**
As discussed in https://github.com/TextureGroup/Texture/pull/1190, let's try smaller change for now.

As a follow up on https://github.com/TextureGroup/Texture/issues/1045 I decided it's time to do this kinda complicated change.

More than two years after @garrettmoon introduced on Medium that AsyncDisplayKit is now Texture [here](https://medium.com/@Pinterest_Engineering/introducing-texture-a-new-home-for-asyncdisplaykit-e7c003308f50), it's time to rename modules, app itself, project and CocoaPods pod.

This PRs tries to do that.

**The king is dead, long live the king!**

<hr>
Also included:
- migrate all Swift projects to Swift 4.2 as Swift 3 (or Swift 2 in some cases!) is not compatible with latest XCode anymore (Swift 2 examples are not compatible even with XCode 9)
- fix podfiles of some projects
- fix project settings and imports of some projects

## NOTE:
Let's please move on on merging as outdated examples help no one. :)

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/f2420b528bb7bf068f37d227dd06f7ea/raw/20f8fe7e9009e6b4a179fcf30c0ad04ec7fac244/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/42cdf7d0446a1f4a6b42e8265c97ad06/raw/36382acc243ec1e4e3c1a7b59f8c35cf81bd29af/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/e937f7023577b79b6942e2212576fce3/raw/023e9be5d949b501a6655c2101fca3f17b3e8049/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/868f44004fe30e69a4f74579a6fcf4e4/raw/33fc069e557f43cd1356d35b3eb94e3cbcf13e43/log.txt)

**Comment by Kaspik** [comment]
@garrettmoon Updated probably all example projects. Is it possible to run the CI now so I can see what's missing?

Also, it;s probably ready for review now! :)

**Comment by Kaspik** [comment]
Created another PR (https://github.com/TextureGroup/Texture/pull/1450) with all examples migrated to latest XCode and fixed.

Let's first get this to master so I can continue there as well please. :)

**Comment by Kaspik** [comment]
@Adlai-Holler / @nguyenhuy / @garrettmoon / @maicki 
Any update on this guys please? I would like to move this before there are another conflicts. :)

**Comment by Adlai-Holler** [comment]
The diff right now is too aggressive, for instance it changes the name of internal methods like asyncdisplaykit_isAsyncTransactionContainer -> texture_isAsyncTransactionContainer.

For 3.0, how about we rename just the library and the umbrella header? That way developers won’t have to change any executable code, just their import statements.

It sucks considering how much work you've put into it, but it's just not an option to land this as-is. Consider the fact that although NextStep was discontinued 24 years ago, Apple's Foundation library still uses its NS prefix.

**Comment by Kaspik** [comment]
@Adlai-Holler Hehe, I just thought you are on the previous PR and not this new one...

Yeah, I did this based on the old convo so it's not aggressive or anything like that (I agree it was aggressive), so this should be just about changing `<AsyncDisplayKit/...>` to `<Texture/...>` for imports. :)

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/contents/examples/AsyncDisplayKitOverview/Sample/Node%2520Containers/OverviewASCollectionNode.h?ref=9675e0fddfa853df6594b924bf69278f50e67d07: 404 - Not Found // See: https://developer.github.com/v3/repos/contents/#get-contents. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.6

```
 #  from Dangerfile:43
 #  -------------------------------------------
 #        data = ""
 >        contents = github.api.get file["contents_url"]
 #        open(contents["download_url"]) { |io|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/b4f851b0eb7a0f43ad53259cdf167bd6/raw/c336dbc68f715ee3407be6ada38a5d9085048a55/log.txt)

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/contents/examples/AsyncDisplayKitOverview/Sample/Node%2520Containers/OverviewASCollectionNode.h?ref=e96763455ba6b93b88b036c9c5d74574c081789b: 404 - Not Found // See: https://developer.github.com/v3/repos/contents/#get-contents. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.6

```
 #  from Dangerfile:43
 #  -------------------------------------------
 #        data = ""
 >        contents = github.api.get file["contents_url"]
 #        open(contents["download_url"]) { |io|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/576cd99885529fe03b1ad443f5035df6/raw/5fd38091243d1d7379623d824b8f5184585af613/log.txt)

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/contents/examples/AsyncDisplayKitOverview/Sample/Node%2520Containers/OverviewASCollectionNode.h?ref=5c64c0a63d00bc90d45d1a095bbefddc2e0fb57b: 404 - Not Found // See: https://developer.github.com/v3/repos/contents/#get-contents. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.6

```
 #  from Dangerfile:43
 #  -------------------------------------------
 #        data = ""
 >        contents = github.api.get file["contents_url"]
 #        open(contents["download_url"]) { |io|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/614cfe3cb8f3fcdc112c5843ef462f28/raw/17aeccf453a7f00b71683b84519218462128a894/log.txt)

**Comment by Kaspik** [comment]
Good morning. I feel like there is something wrong with CI. It uses Cocoapods 1.1.0 instead of 1.6.1., it fails on github error 404 (like it wouldn't be related to my changes) and I can't seem to find a solution.

How old is the CI setup? Should we update it?

**Comment by nguyenhuy** [comment]
ah yeah, there seems to be a mismatch in CocoaPods versions. The CI is basically a Mac mini machine sitting close to us. We can try to update CocoaPods, but is 1.6.1 really needed for this diff?

cc @garrettmoon.

**Comment by Kaspik** [comment]
@nguyenhuy 1.6.1 is default for all my projects so I didn't do anything special - we can use any other version, but I don't think we should use something lower than more than year old version 1.5.0.

Anyways, I don't think it's the problem why it fails, and I'm not able to find out what really is based on the error message...

@garrettmoon Updating the machine would be possible? Is there something I can help with even if it should be another diff?

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/contents/examples/AsyncDisplayKitOverview/Sample/Node%2520Containers/OverviewASCollectionNode.h?ref=7e58bbba25aefb4b5acdb1db1e936534d01d8098: 404 - Not Found // See: https://developer.github.com/v3/repos/contents/#get-contents. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.6

```
 #  from Dangerfile:43
 #  -------------------------------------------
 #        data = ""
 >        contents = github.api.get file["contents_url"]
 #        open(contents["download_url"]) { |io|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/f69b4ad5047d9fc08789f1fedb4c8b3c/raw/62b13b67d17f2fcff26d1c0f52cb33b8aa892b8f/log.txt)

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/pulls/1416/files?per_page=100: 502 - Server Error. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.6

```
 #  from Dangerfile:37
 #  -------------------------------------------
 #    pr_number = github.pr_json["number"]
 >    files = github.api.pull_request_files(repo_name, pr_number)
 #    files.each do |file|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/da6bfeb55c7fd0cdc5265c9b82d11d40/raw/eb0bded5dce6535317165f975e54f32a6249487d/log.txt)

**Comment by Kaspik** [comment]
@nguyenhuy @Adlai-Holler Hey guys, solved another conflicts. It's getting harder and harder to keep this on track, but I can't figure the CI. Can anyone, like anyone please, try to tell me whatsup? I can't figure it out.

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/contents/Source/ASViewController.h?ref=5f652db794fdd58a1a16ba5a5c216938e5ae0ec0: 502 - Server Error. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.6

```
 #  from Dangerfile:43
 #  -------------------------------------------
 #        data = ""
 >        contents = github.api.get file["contents_url"]
 #        open(contents["download_url"]) { |io|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/976ccfa3084ed3318b19afc17731d6db/raw/270156f787682c7864198f77b2d883d685d40c2f/log.txt)

**Comment by nguyenhuy** [comment]
@Kaspik: @garrettmoon will look into the CI and see if we can upgrade its CocoaPods installation to a newer version (1.6.0 or 1.6.1, assuming the latter will make it easier for you).

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/contents/examples/AsyncDisplayKitOverview/Sample/Node%2520Containers/OverviewASCollectionNode.h?ref=48659b0c626848aad465baec90a7d16c7f9cc2f9: 404 - Not Found // See: https://developer.github.com/v3/repos/contents/#get-contents. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.6

```
 #  from Dangerfile:43
 #  -------------------------------------------
 #        data = ""
 >        contents = github.api.get file["contents_url"]
 #        open(contents["download_url"]) { |io|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/7394c1fed2a4506cfc52fd6e3cda1955/raw/031e97a6016d4c5afc026fc0726e2a17aeae4c48/log.txt)

**Comment by Kaspik** [comment]
@nguyenhuy / @garrettmoon Hello guys! Any update on this please? :) Just merged master so we will see but I wanted to make sure it's in progress so we can merge this.

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/contents/examples/AsyncDisplayKitOverview/Sample/Node%2520Containers/OverviewASCollectionNode.h?ref=1f9cb29f2c1a4872a9f4d925c72074784fd9df65: 404 - Not Found // See: https://developer.github.com/v3/repos/contents/#get-contents. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.7

```
 #  from Dangerfile:43
 #  -------------------------------------------
 #        data = ""
 >        contents = github.api.get file["contents_url"]
 #        open(contents["download_url"]) { |io|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/7125e96cb8fcd5ad382e3173d01a8001/raw/98616e2741210e83ce0443800e6b5c8e9795a619/log.txt)

**Comment by pinterest-ios** [comment]
## Danger has errored
[!] Invalid `Dangerfile` file: GET https://api.github.com/repos/TextureGroup/Texture/contents/examples/AsyncDisplayKitOverview/Sample/Node%2520Containers/OverviewASCollectionNode.h?ref=e8226130df3658410a2b7ca385c1258a1afb0dd0: 404 - Not Found // See: https://developer.github.com/v3/repos/contents/#get-contents. Updating the Danger gem might fix the issue. Your Danger version: 5.5.10, latest Danger version: 6.0.7

```
 #  from Dangerfile:43
 #  -------------------------------------------
 #        data = ""
 >        contents = github.api.get file["contents_url"]
 #        open(contents["download_url"]) { |io|
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/e8a3f447ca1b8bf0f531e645cba03c94/raw/1bf727ee925492e9c2154ff19299b0dc12972543/log.txt)

**Comment by Kaspik** [comment]
Apparently there is no point in supporting this project anymore. You don't really care about PRs by community, releases happen once per year and your Slack activity is not really user-friendly... I'm sad about that, but if you don't want to spend time on these changes made by community, I'm not gonna spend more time on these PRs... 😞

**Comment by Kaspik** [comment]
Also, there is still second PR that fixes all about ~50 example projects and migrates them from Swift 2 and 3 to Swift 4.2 - https://github.com/TextureGroup/Texture/pull/1450

I believe it would be helpful to merge them both.

**Comment by Kaspik** [comment]
@bolsinga I saw you are recently the most active... are you able to move on on this PR? It's approved by two other guys, Danger fails incorrectly (I'll revert to ruby danger tomorrow as the JS one is wrong and not working where the ruby is working with small update) and everything else is green...

**Comment by nguyenhuy** [comment]
@Kaspik I’m on leave right now and won’t be back until late this year. As such, I’ll defer to other maintainers on this PR. And thank you for following up on it. Admittedly we have been dropping the ball on this one which causes a bad experience to you. I’m personally sorry about that. The core maintainers were debating about the ideal strategy to release 3.0 in which this PR is intended to be. However, at this point, in my personal opinion, we should just land this PR and call the next release 3.0.

**Comment by nguyenhuy** [comment]
On a somewhat related note, I think we can all benefit from a more frequent and predictable release schedule, i.e a monthly release schedule. To make it easier to make a release, we may use GitHub Actions to automate some of the steps of the [release process](https://github.com/TextureGroup/Texture/blob/master/RELEASE.md).

**Comment by Kaspik** [comment]
Okay, as Huy is out ( 😭 ), and it feels like Garrett and Adlai are not really responding (either here or on Slack), let me try to ping few more people that seem to be active these days. (🙏)

cc @rahul-malik @bolsinga 

Hey guys! 👋 This PR contains whole history and discussion, links to previous PR and Issues and also some info why it's open for this long (we waited for new CI and for 2.8.1 release).
Now it's ready to be merged and it's really just about hitting the `Merge` button - nothing else.

It's up to you if/when you want to release 3.0 then, but I don't want to keep resolving merge conflicts on this PR... so please! ❤️

File: Source/Texture.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
+//
+//  AsyncDisplayKit.h
```
Comment (bolsinga) [inline Source/Texture.h]: this name ought to change.

**Comment by Kaspik** [comment]
@bolsinga But the nice thing about the way we chose is that you can find and replace `#import <AsyncDisplayKit/` with `#import <Texture/`, right? And we all kinda expected that when Garret announced that two years ago...

File: Source/Texture.h (hunk 1 above)
Comment (Kaspik) [reply Source/Texture.h]: Sorry, resolved already, I was out.

**Comment by Kaspik** [comment]
Everything up-to-date.

@bolsinga This solution was the "chosen" one / agreed on - note here: https://github.com/TextureGroup/Texture/pull/1190#issuecomment-474519182

Also agreed on with other Texture members - I opened this PR before 2.8.0 Texture release - and as Garrett said - this needs to wait for 2.8.0 release and then it's good to go to master for 3.0.0 release:
Screen Shot 2019-09-16 at 15 12 52

I believe it's easier, as I mentioned, to do find-n-replace across whole project that you want to upgrade to 3.0.0 than doing it just partially.
It's also _expected_ that major upgrade might come with breaking changes - and honestly, with current release cycle, if someone is not using Texture from master, but from CocoaPods, he'll need to do it anyways. Try to compare [Texture 2.8 vs. current master](https://github.com/TextureGroup/Texture/compare/2.8.1...master?expand=1).

That being said - I think it could and _should_ go to master as it is ideally. 🙂

**Comment by rahul-malik** [comment]
Hey @Kaspik - Thanks so much for the contribution and your patience. We're actively discussing a plan around coordinating 3.0 and a final 2.x release and should be getting back to you shortly with more details. In the meanwhile, I think it would be helpful to have a simple shell script in the repository that yourself and others could run to do the import renaming if you're up for it.

**Comment by Kaspik** [comment]
@rahul-malik Hey! It has been month since your last response, couple of merge conflicts, couple problems with CI... How does it look?

**Comment by rahul-malik** [comment]
@Kaspik - Hey! Sorry didn't see your response here. We're actively trying to get 2.9 released and then we will take this change (and potentially a couple others) and release Texture as 3.0. I'll let you know when to rebase / update your PR and thank you for all your work :)

**Comment by Kaspik** [comment]
Good morning.

After months of refactoring, we removed Texture from our project last week (this was bad experience for me, as well as not really regular releases and updates for new iOS versions etc.)
I'm not going to maintain this anymore so feel free to fork my repo if you wanna keep it and finish it. :)

**Pull request closed without merging**